packages/gated-content/src/__tests__/GatedClient.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { Wallet } from 'ethers';
+
+import { GatedClient } from '../GatedClient';
+import { testing } from '../__helpers__/env';
+import { webCryptoProvider } from '../web';
+
+const ownerId = mockProfileId();
+
+const metadata = mockPublicationMetadata({
+  content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
+});
+const signer = Wallet.createRandom();
+
+const eoaOwnershipAccessCondition = mockEoaOwnershipAccessCondition({
+  address: signer.address,
+});
+const profileOwnershipAccessCondition = mockProfileOwnershipAccessCondition({ profileId: ownerId });
+const accessCondition = mockOrAccessCondition([
+  profileOwnershipAccessCondition,
+  eoaOwnershipAccessCondition,
+]) as AccessCondition;
+
+function setupTestScenario() {
+  const client = new GatedClient({
+    authentication: {
+      domain: 'localhost',
+      uri: 'https://localhost/login',
+    },
+    signer,
+    environment: testing,
+    storageProvider: new InMemoryStorageProvider(),
+    encryptionProvider: webCryptoProvider(),
+  });
+
+  return { client };
+}
+
+describe(`Given an instance of the ${GatedClient.name}`, () => {
+  describe(`when calling the "${GatedClient.prototype.encryptPublication.name}" method`, () => {
+    it(`should return the expected GatedPublicationMetadata`, async () => {
+      const { client } = setupTestScenario();
+
+      const result = await client.encryptPublication(metadata, accessCondition);
+
+      expect(result.unwrap()).toMatchObject({
+        ...metadata,
+
+        content: 'This publication is gated.',
+
+        encryptionParams: {
+          accessCondition,
+          providerSpecificParams: {
+            encryptionKey: expect.any(String),
+          },
+          encryptionProvider: EncryptionProvider.LitProtocol,
+          encryptedFields: {
+            content: expect.any(String),
+          },
+        },
+      });
+    });
+  });
+
+  describe(`when calling the "${GatedClient.prototype.decryptPublication.name}" method`, () => {
+    it(`should be able to decrypt and return the expected MetadataFragment`, async () => {
+      // setup
+      const { client } = setupTestScenario();
+      const encryptionResult = await client.encryptPublication(metadata, accessCondition);
+      const encrypted = encryptionResult.unwrap();
+
+      // simulate the transcription process done by the indexer and Lens API
+      const metadataFragment = mockMetadataFragment({
+        content: encrypted.content,
+      });
+      const encryptionParams = mockEncryptionParamsFragment({
+        ownerId,
+        others: [eoaOwnershipAccessCondition],
+        encryptionKey: encrypted.encryptionParams.providerSpecificParams.encryptionKey,
+        encryptedFields: mockEncryptedFieldsFragment(
+          encrypted.encryptionParams.encryptedFields as EncryptedFieldsFragment,
+        ),
+      });
+
+      // exercise
+      const decryptionResult = await client.decryptPublication(metadataFragment, encryptionParams);
```

Comment:
For now it seems there is not really a use case to support decryption of raw Publication Metadata (i.e. the `GatePublicationMetadata` returned by the `encryptPublication` method).

From the `@lens-protocol/react` SDK perspective the usage of `GatedClient` is completely segregated and makes use of the same data structure used by `CommentFragment` and `PostFragment` (i.e. any fragment shape of `MetadataOutput`). This has  the nice advantage of not requiring a bespoke render path (nor transformation within the SDK) for rendering decrypted gated content. It also does not require to make an extra request to download raw metadata from `contentURI` (see how Lenster implemented token-gated to get an idea of these pain points).

`GatedClient` will also receive a small adapter to work within the low-level client SDK (i.e. `@lens-protocol/client`). This will be type safe in the sense that the integrator will be guided by the TS typedef to feed the entire `CommentFragment` and `PostFragment` into the adapter interface. The adapter will then use this `GatedClient#decryptPublication` as needed.

If the need arise to be able to decrypt raw metadata (as per JSON file uploaded on `contentURI`), I strongly suggest to create a bespoke method that knows the details of that data structure. As comments left in the code, there are some discrepancies between raw data (name, camelCases vs. snake_cased properties, missing properties). All these are implementation details, a consumer using the Lens SDK does not even need to know about these, so we can improve the Lens API over time and fix those discrepancies (if need to).